src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { ErrorIcon } from "../../assets/badge/ErrorIcon";
+import { SuccessIcon } from "../../assets/badge/SuccessIcon";
+import { WarningIcon } from "../../assets/badge/WarningIcon";
+import { BadgeContainer } from "./Badge.css";
+import { Body } from "../Typography";
+
+type BadgeProps = {
+  children: React.ReactNode;
+  icon?: React.ReactNode;
+  status?: "success" | "error" | "warning";
+  theme?: "light" | "dark";
+};
+
+type StatusIcon = {
+  error: React.ReactNode;
+  success: React.ReactNode;
+  warning: React.ReactNode;
+};
```

Comment:
```suggestion
type StatusIcon = {
 [key in BadgeProps['status']]: React.ReactNode
};
```
@Serse2 what do you think?

Reply:
Good point! i change che type like this

`type Status = "success" | "error" | "warning";`

```
type StatusIcon = {
  [key in Status]: React.ReactNode;
};
```